core/modules/simpletest/simpletest.module

```diff
@@ -369,16 +369,16 @@ function simpletest_test_get_all() {
         $classes = backdrop_parse_info_file($file->uri, TRUE);
         $error = '';
         if (empty($classes)) {
-          $error = 'File %uri contains no testing classes.';
+          $error = t('File @uri contains no testing classes.');
         }
         else {
+          // If the .tests.info file is malformed, it puts a slew of
+          // hard-to-diagnose errors into the watchdog log when the testing page
+          // is loaded. So we'll check for various malformalities and put a more
+          // informative error.
           foreach ($classes as $class => $info) {
-            // If the .tests.info file is malformed, it puts a slew of
-            // hard-to-diagnose errors into the watchdog log when the testing page
-            // is loaded. So we'll check for various malformalities and put a more
-            // informative error.
             if (!is_array($info) || empty($info)) {
-              $error = format_string('File %uri contains a definition for %class that is not an array of testing class fields.', array('%class' => $class));
+              $error = t('File @uri contains a definition for @class that is not an array of testing class fields.', array('@class' => $class));
```

Comment:
Same question as above - should that `@uri` get replaced?

Reply:
Alll `@uri` (or `%uri`) are replaced later on by `format_string`. But there is a bigger issue here. The [D7 documentation](https://www.drupal.org/docs/7/testing/simpletest-testing-tutorial-drupal-7#when-to-use-t) lists very few cases when `t()` is supposed to be used. This is not one of those. My suggestion is to stick with `format_string` here.

Reply:
In either case, it's strange to have t() for some of these and format_string later on. So, if we are going to use t(), then I agree that it's best to replace uri  each time it's used.

Reply:
OK.

Reply:
I still find it confusing that different replacements happen in different places, but if that's unavoidable ... :shrug:

Reply:
I tried something different in the new PR with all replacements in one place. https://github.com/backdrop/backdrop/pull/4249